cognite/neat/rules/models/rules/_dms_architect_rules.py

```diff
@@ -849,40 +850,18 @@ def _create_views_with_node_types(
         node_types = dm.NodeApplyList([])
         parent_views = {parent for view in views for parent in view.implements or []}
         for view in views:
-            ref_containers = sorted(view.referenced_containers(), key=lambda c: c.as_tuple())
             dms_view = dms_view_by_id.get(view.as_id())
-            has_data = dm.filters.HasData(containers=list(ref_containers)) if ref_containers else None
-            if dms_view and isinstance(dms_view.reference, ReferenceEntity):
-                # If the view is a reference, we implement the reference view,
-                # and need the filter to match the reference
-                ref_view = dms_view.reference.as_view_id()
-                node_type = dm.filters.Equals(
-                    ["node", "type"], {"space": ref_view.space, "externalId": ref_view.external_id}
-                )
-            else:
-                node_type = dm.filters.Equals(["node", "type"], {"space": view.space, "externalId": view.external_id})
-            if view.as_id() in parent_views:
-                if dms_view and dms_view.filter_ == "nodeType":
+            view.filter = self._create_view_filter(view, dms_view, parent_views)
+            if (
+                isinstance(view.filter, dm.filters.Equals)
+                and isinstance(view.filter._value, dict)
+                and (node_space := view.filter._value.get("space"))
+                and (node_ext_id := view.filter._value.get("externalId"))
+            ):
+                node_types.append(dm.NodeApply(space=node_space, external_id=node_ext_id))
+                if view.as_id() in parent_views:
                     warnings.warn(issues.dms.NodeTypeFilterOnParentViewWarning(view.as_id()), stacklevel=2)
```

Comment:
I guess this gets brought up in `issues` when importing rules?

Reply:
Yes.